src/components/OpenWith/OpenWith.js

```diff
@@ -192,9 +204,69 @@ class OpenWith extends PureComponent<Props, State> {
      * @private
      * @return {void}
      */
-    onIntegrationClick(): void {
-        /* no-op */
-    }
+    onIntegrationClick = (appIntegrationId: string): void => {
+        const { fileId }: Props = this.props;
+        this.api
+            .getAppIntegrationsAPI(false)
+            .execute(
+                appIntegrationId,
+                fileId,
+                this.executeIntegrationSuccessHandler,
+                this.executeIntegrationErrorHandler,
+            );
+    };
+
+    /**
+     * Opens the integration in a new tab based on the API data
+     *
+     * @private
+     * @param {ExecuteAPI} executePostData - API response on how to open an executed integration
+
+     * @return {void}
+     */
+    executeIntegrationSuccessHandler = (executePostData: ExecuteAPI): void => {
+        const { method, url } = executePostData;
+        switch (method) {
+            case HTTP_POST:
+                this.setState({ executePostData });
+                break;
+            case HTTP_GET:
+                // Use a unique combination of integration + file ID to avoid creation of new windows
+                const windowRef = window.open(url);
```

Comment:
Is the above comment correct? I am assuming you are meaning the 2nd param to window.open

Reply:
whoops, I need to remove that since we aren't doing any optimizations with window opening yet.